src/tainting/Taint.ml

```diff
@@ -84,11 +85,32 @@ let length_of_call_trace ct =
 
 type sink = { pm : Pattern_match.t; rule_sink : R.taint_sink } [@@deriving show]
 
+let compare_metavar_env env1 env2 =
+  (* It's important that we only return 0 if the two bindings are
+     structurally equal. Otherwise, there will be many duplicates.
+     We use Stdlib.compare in the other case because deriving
+     compare is rather difficult and the specific ordering doesn't
+     matter. *)
+  if Metavariable.Structural.equal_bindings env1 env2 then 0
```

Comment:
What if we had a `Set` module for `Metavariable.bindings` and used it here ? `Set` has a `compare` already implemented and it would not be sensitive to ordering.

Reply:
Wouldn't we need to also write the compare function for these tests?